veloxms-core/src/main/scala/edu/berkeley/veloxms/VeloxApplication.scala

```diff
@@ -92,9 +94,53 @@ class VeloxApplication extends Application[VeloxConfiguration] with Logging {
     jsonMapper.readValue(json, classOf[Seq[String]])
   }
 
+  def createModel(name: String,
+                  sparkContext: SparkContext,
+                  etcdClient: EtcdClient,
+                  broadcastProvider: BroadcastProvider,
+                  sparkDataLocation: String,
+                  partitionMap: Seq[String],
+                  env: Environment,
+                  hostname: String
+                  ): Unit = {
+
+    val key = s"$configEtcdPath/models/$name"
+    val json = etcdClient.getValue(key)
+    val jsonTree = jsonMapper.readTree(json)
+    require(jsonTree.has("modelType"))
+    require(jsonTree.has("onlineUpdateDelayInMillis"))
+    require(jsonTree.has("batchRetrainDelayInMillis"))
+    val modelType = jsonTree.get("modelType").asText
+    val onlineUpdateDelay = jsonTree.get("onlineUpdateDelayInMillis").asLong
+    val batchRetrainDelay = jsonTree.get("batchRetrainDelayInMillis").asLong
+    val modelSpecificConfig: Option[JsonNode] = if (jsonTree.has("config")) {
+      Some(jsonTree.get("config"))
+    } else {
+      None
+    }
+    val mirror = ru.runtimeMirror(getClass.getClassLoader)
```

Comment:
Yeah. After _a lot_ of trial and error, I ended up being able to use an example in the reflection API docs http://docs.scala-lang.org/overviews/reflection/overview.html (I'll put in a comment about this). The only assumption about the implementation is that it has to take exactly the same constructor arguments as Model.scala, because that is hard-coded into the constructor call. I'm working on a bit of documentation on how to add new models, and I'll discuss these limitations and how to get around them.